File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        ans, n = [0] * len(nums), len(nums) - 1
        l, r = 0, n
        
        while l <= r:
            if abs(nums[l]) > abs(nums[r]):
                ans[n], l = nums[l] ** 2, l + 1
            else:
                ans[n], r = nums[r] ** 2, r - 1
            n -= 1
            
        return ans

File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int n = nums.size();
        int beg = 0, end = n - 1, idx = n - 1;
        vector<int>v(n);
        while (beg <= end) {
            if (nums[beg] * nums[beg] < nums[end] * nums[end]) {
                v[idx--] = nums[end] * nums[end];
                end--;
            }
            else {
                v[idx--] = nums[beg] * nums[beg];
                beg++;
            }
        }
        return v;
    }
};